plugin-non-client-call-control/src/listeners/CustomListeners.js

```diff
@@ -35,3 +37,32 @@ Flex.Actions.addListener('beforeHangupCall', async (payload, abort) => {
   await hangupNonWebRtcCall(task);
   abort();
 });
+
+Flex.Actions.addListener("beforeStopMonitoringCall", async (payload, abort) => {
+  if (FlexState.isWorkerUsingWebRTC()) {
+    return;
+  }
+  const { task } = payload;
+  if (!Flex.TaskHelper.isCallTask(task)) {
+    return;
+  }
+  console.info("stop monitoring call");
+  const conference = task && task.conference;
+  const conferenceSid = conference && conference.conferenceSid;
+  const myWorkerSid = FlexState.getMyWorkerSid();
+  console.log("my worker sid" + myWorkerSid);
+  const conferenceChildren = conference?.source?.children || [];
+  const supervisorParticipant = conferenceChildren.find(
+    (p) =>
+      p.value.participant_type === "supervisor" &&
+      p.value.status === "joined" &&
+      myWorkerSid === p.value.worker_sid
+  );
+  console.log(`Current supervisorSID = ${supervisorParticipant.key}`);
+  const supervisorParticipantSid = supervisorParticipant.key;
```

Comment:
@khu-twilio What do you think of adding a conditional check here, in case there's no matching participant and supervisorParticipant is undefined? If undefined, there's no need to attempt the function call which would also return an error.